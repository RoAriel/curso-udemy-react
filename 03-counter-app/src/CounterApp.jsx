import { useState } from "react"
import PropTypes from "prop-types"

export const CounterApp = ({ value }) => {

    const [count, setCount] = useState(value)

    const handleAdd = () => { setCount(count + 1) }
    const handleSubstract = () => { setCount(count - 1) }
    const handleRest = () => { setCount(value) }

    return (
        <>
            <h1>Counter App</h1>
            <h2>{count}</h2>

            <button aria-label='btn-add' onClick={handleAdd}> +1 </button>
            <button aria-label='btn-subst' onClick={handleSubstract}> -1 </button>
            <button aria-label='btn-reset' onClick={handleRest}> Reset </button>
        </>
    )
}

CounterApp.prototype = {
    value: PropTypes.number.isRequired,
}