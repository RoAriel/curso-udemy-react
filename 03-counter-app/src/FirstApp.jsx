import PropTypes from "prop-types"

export const FirstApp = ({ title = 'No title', subTitle = 'No sub', name }) => {

    return (
        <>
            <h1 data-testid='test-title'>{title}</h1>
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            <p>{name}</p>
        </>
    )
}

FirstApp.prototype = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}
