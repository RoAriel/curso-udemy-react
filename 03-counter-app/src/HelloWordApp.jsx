export const HelloWordApp = () => {
  return (
    <h1>
      Hello World App
    </h1>
  )
}

