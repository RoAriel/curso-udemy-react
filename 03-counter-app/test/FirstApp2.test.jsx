import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp />', () => {

    const title = 'Soy un title'
    const sub = 'Soy un sub'

    test('should match con snapshot', () => {

        const { container } = render(<FirstApp title={title} />)

        expect(container).toMatchSnapshot()
    })

    test('should  mostrar msj "Soy un title"', () => {

        render(<FirstApp title={title} />)
        expect(screen.getByText(title)).toBeTruthy()
        //screen.debug()
    })

    test('should de tener el titulo en h1', () => {

        render(<FirstApp title={title} />)

        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)

    })

    test('should mostrar subs enviado por props', () => {
        render(
            <FirstApp
                title={title}
                subTitle={sub}
            />)

        expect(screen.getAllByText(sub).length).toBe(2)

    })

})
