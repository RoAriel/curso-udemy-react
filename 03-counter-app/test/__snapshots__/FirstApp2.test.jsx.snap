// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> should match con snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Soy un title
  </h1>
  <p>
    No sub
  </p>
  <p>
    No sub
  </p>
  <p />
</div>
`;
